File: MHikeApp/app/src/main/java/com/example/mhike/models/Hike.java
package com.example.mhike.models;

import java.io.Serializable;

public class Hike implements Serializable {
    private int id;
    private String name;
    private String location;
    private String date;
    private boolean parking;
    private double length;
    private String difficulty;
    private String description;
    private String weather;
    private int groupSize;

    // Constructors
    public Hike() {
    }

    public Hike(String name, String location, String date, boolean parking,
                double length, String difficulty, String description,
                String weather, int groupSize) {
        this.name = name;
        this.location = location;
        this.date = date;
        this.parking = parking;
        this.length = length;
        this.difficulty = difficulty;
        this.description = description;
        this.weather = weather;
        this.groupSize = groupSize;
    }

    // Getters and Setters
    public int getId() {
        return id;
    }

    public void setId(int id) {
        this.id = id;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public String getLocation() {
        return location;
    }

    public void setLocation(String location) {
        this.location = location;
    }

    public String getDate() {
        return date;
    }

    public void setDate(String date) {
        this.date = date;
    }

    public boolean isParking() {
        return parking;
    }

    public void setParking(boolean parking) {
        this.parking = parking;
    }

    public double getLength() {
        return length;
    }

    public void setLength(double length) {
        this.length = length;
    }

    public String getDifficulty() {
        return difficulty;
    }

    public void setDifficulty(String difficulty) {
        this.difficulty = difficulty;
    }

    public String getDescription() {
        return description;
    }

    public void setDescription(String description) {
        this.description = description;
    }

    public String getWeather() {
        return weather;
    }

    public void setWeather(String weather) {
        this.weather = weather;
    }

    public int getGroupSize() {
        return groupSize;
    }

    public void setGroupSize(int groupSize) {
        this.groupSize = groupSize;
    }

    @Override
    public String toString() {
        return "Hike{" +
                "id=" + id +
                ", name='" + name + '\'' +
                ", location='" + location + '\'' +
                ", date='" + date + '\'' +
                ", parking=" + parking +
                ", length=" + length +
                ", difficulty='" + difficulty + '\'' +
                '}';
    }
}

File: mhikeflutteref/lib/models/hike.dart
class Hike {
  int? id;
  String name;
  String location;
  String date;
  bool parking;
  double length;
  String difficulty;
  String? description;
  String? weather;
  int? groupSize;

  Hike({
    this.id,
    required this.name,
    required this.location,
    required this.date,
    required this.parking,
    required this.length,
    required this.difficulty,
    this.description,
    this.weather,
    this.groupSize,
  });

  Map<String, dynamic> toMap() {
    return {
      'id': id,
      'name': name,
      'location': location,
      'date': date,
      'parking': parking ? 1 : 0,
      'length': length,
      'difficulty': difficulty,
      'description': description,
      'weather': weather,
      'groupSize': groupSize,
    };
  }

  factory Hike.fromMap(Map<String, dynamic> map) {
    return Hike(
      id: map['id'],
      name: map['name'],
      location: map['location'],
      date: map['date'],
      parking: map['parking'] == 1,
      length: map['length'],
      difficulty: map['difficulty'],
      description: map['description'],
      weather: map['weather'],
      groupSize: map['groupSize'],
    );
  }

  Hike copyWith({
    int? id,
    String? name,
    String? location,
    String? date,
    bool? parking,
    double? length,
    String? difficulty,
    String? description,
    String? weather,
    int? groupSize,
  }) {
    return Hike(
      id: id ?? this.id,
      name: name ?? this.name,
      location: location ?? this.location,
      date: date ?? this.date,
      parking: parking ?? this.parking,
      length: length ?? this.length,
      difficulty: difficulty ?? this.difficulty,
      description: description ?? this.description,
      weather: weather ?? this.weather,
      groupSize: groupSize ?? this.groupSize,
    );
  }
}
